irods_specific_implementation/indexed_connection_pool_with_expiry.hpp

```diff
@@ -214,12 +215,30 @@ namespace irods {
             return payload[keyword::user_name].as_string();
         } // get_user_name_from_key
 
+        auto save_rodsadmin_password_if_necessary() -> void
+        {
+            namespace irc = irods::rest::configuration;
+
+            if (const int ec = obfGetPw(nullptr); ec < 0) {
+                const auto& rodsadmin_password = irc::irods_client_environment().at("rodsadmin_password").get_ref<const std::string&>();
+                if (const int ec = obfSavePw(0, 0, 0, rodsadmin_password.data()); ec < 0) {
+                    THROW(ec, "Failed to save password for rodsadmin proxy user");
+                }
+            }
+        } // save_rodsadmin_password_if_necessary
+
         auto make_connection(const std::string& _jwt) -> std::shared_ptr<connection_handle>
         {
+            // TODO: Need to recognize/parse zone names here for remote use cases?
```

Comment:
I think including a zone name could potentially break this, but haven't tried it yet

Reply:
part of that new issue

Reply:
Done